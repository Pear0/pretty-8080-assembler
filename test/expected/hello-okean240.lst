	                        ; 🐟 для Океана-240
	                        ; Для загрузки в монитор через директиву L
	                        ; Автор: Тимур Ташпулатов, C.-Петербург 2018
	                	.project okeah
	                	.org 8000h
	                
	                BANKING	equ	0c1h		; регистр управления банками ОЗУ и ПЗУ
	                VIDEO	equ	0e1h		; регистр управления цветом и режимами видео
	                
8000	3E 01           	mvi	a, 01
8002	D3 C1           	out	BANKING		; включить отображение старших 32К на младшие
	                
8004	AF              	xra	a		; очистить аккумулятор
8005	21 00 40        	lxi	h, 4000h	; поместить в HL адрес начала экранной области
8008	16 00           	mvi	d, 0		; очистить регистр D можно было бы на один байт короче
	                
800A	01 00 20        	lxi	b, 256*32	; экран в "Океане" состоит из вертикальных столбиков по 256 байт в каждом
800D	72              Loop:	mov	m, d		; заполняем экран трухой
800E	14              	inr	d		; примерно на треть
800F	23              	inx 	h		; двигаться дальше
	                
8010	0B              	dcx 	b		; внимательно следим за длиной столбика
8011	78              	mov	a, b		; если столбик 
8012	B1              	ora	c		; не дорисован
8013	C2 0D 80        	jnz	Loop		; продолжаем писать в экран труху
	                
8016	21 00 41        	lxi	h, 4100h	; цвет каждой точки определяется битами соседних через 100h байт
8019	3E 00           	mvi	a, 00h		; первый столбик
801B	CD 53 80        	call	Strip		; окрашиваем нулём
	                
801E	21 00 43        	lxi	h, 4300h	; второй столбик
8021	3E 0F           	mvi	a, 0fh		; окрашиваем в две полосы
8023	CD 53 80        	call	Strip		; для красоты
	                
8026	21 00 45        	lxi	h, 4500h	; третье окно
8029	3E 33           	mvi	a, 33h		; выходит
802B	CD 53 80        	call	Strip		; к океану
	                
802E	21 00 47        	lxi	h, 4700h	; ровным ветром
8031	3E 55           	mvi	a, 55h		; дышит
8033	CD 53 80        	call	Strip		; океан
	                
8036	21 00 49        	lxi	h, 4900h	; а за ним
8039	3E FF           	mvi	a,0ffh		; диковинные
803B	CD 53 80        	call	Strip		; страны
	                
	                				; и никто
	                				; не видел
	                				; этих стран
	                
803E	AF              	xra	a		; экран размалеван
803F	D3 C1           	out	BANKING		; возвращаем его на место
	                
8041	3C              Key:	inr	a		; не пропадать же аккумулятору
8042	E6 3F           	ani	3fh		; возьмем из него три младших бита переднего плана
8044	F6 40           	ori	40h		; и три старших бита фона
8046	D3 E1           	out	VIDEO		; и намажем на кадр
	                
8048	CD 09 E0        	call	0e009h		; подождем ввода символа с клавиатуры
804B	FE 1B           	cpi	1bh		; если это не Esc,
804D	C2 41 80        	jnz	Key		; продолжим интерактивную раскраску
	                
8050	C3 03 E0        	jmp	0e003h		; теплый старт "Монитора"
	                
8053	77              Strip:	mov	m, a		; это подпрограмма
8054	23              	inx	h		; небыстрого, но верного
8055	15              	dcr	d		; рисования столбика
8056	C2 53 80        	jnz	Strip		; с заданною длиною
	                
8059	C9              	ret			; К О Н Е Ц
	                
Labels:
banking                 00C1
key                     8041	loop                    800D
strip                   8053	video                   00E1

Memory dump:
 
8000:  3E 01 D3 C1 AF 21 00 40-16 00 01 00 20 72 14 23  >....!.@.....r.#
8010:  0B 78 B1 C2 0D 80 21 00-41 3E 00 CD 53 80 21 00  .x....!.A>..S.!.
8020:  43 3E 0F CD 53 80 21 00-45 3E 33 CD 53 80 21 00  C>..S.!.E>3.S.!.
8030:  47 3E 55 CD 53 80 21 00-49 3E FF CD 53 80 AF D3  G>U.S.!.I>..S...
8040:  C1 3C E6 3F F6 40 D3 E1-CD 09 E0 FE 1B C2 41 80  .<.?.@........A.
8050:  C3 03 E0 77 23 15 C2 53-80 C9                    ...w#..S........
