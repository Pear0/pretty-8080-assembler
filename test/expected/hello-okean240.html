<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><link href="list-test.css" rel="stylesheet" type="text/css" media="screen"/><link href="../list-test.css" rel="stylesheet" type="text/css" media="screen"/></head><body><pre id="l0"><span class="adr"></span>	                        <span class="cmt">; üêü –¥–ª—è –û–∫–µ–∞–Ω–∞-240</span></pre>
<pre id="l1"><span class="adr"></span>	                        <span class="cmt">; –î–ª—è –∑–∞–≥—Ä—É–∑–∫–∏ –≤ –º–æ–Ω–∏—Ç–æ—Ä —á–µ—Ä–µ–∑ –¥–∏—Ä–µ–∫—Ç–∏–≤—É L</span></pre>
<pre id="l2"><span class="adr"></span>	                        <span class="cmt">; –ê–≤—Ç–æ—Ä: –¢–∏–º—É—Ä –¢–∞—à–ø—É–ª–∞—Ç–æ–≤, C.-–ü–µ—Ç–µ—Ä–±—É—Ä–≥ 2018</span></pre>
<pre id="l3"><span class="adr"></span>	                	<span id="code3">.project okeah</span></pre>
<pre id="l4"><span class="adr"></span>	                	<span id="code4">.org 8000h</span></pre>
<pre id="l5"><span class="adr"></span>	                </pre>
<pre id="l6"><span class="adr"></span>	                <span class="l" id="label6">BANKING</span>	<span id="code6">equ	0c1h		</span><span class="cmt">; —Ä–µ–≥–∏—Å—Ç—Ä —É–ø—Ä–∞–≤–ª–µ–Ω–∏—è –±–∞–Ω–∫–∞–º–∏ –û–ó–£ –∏ –ü–ó–£</span></pre>
<pre id="l7"><span class="adr"></span>	                <span class="l" id="label7">VIDEO</span>	<span id="code7">equ	0e1h		</span><span class="cmt">; —Ä–µ–≥–∏—Å—Ç—Ä —É–ø—Ä–∞–≤–ª–µ–Ω–∏—è —Ü–≤–µ—Ç–æ–º –∏ —Ä–µ–∂–∏–º–∞–º–∏ –≤–∏–¥–µ–æ</span></pre>
<pre id="l8"><span class="adr"></span>	                </pre>
<pre id="l9"><span class="adr">8000</span>	3E 01           	<span id="code9">mvi	a, 01</span></pre>
<pre id="l10"><span class="adr">8002</span>	D3 C1           	<span id="code10">out	BANKING		</span><span class="cmt">; –≤–∫–ª—é—á–∏—Ç—å –æ—Ç–æ–±—Ä–∞–∂–µ–Ω–∏–µ —Å—Ç–∞—Ä—à–∏—Ö 32–ö –Ω–∞ –º–ª–∞–¥—à–∏–µ</span></pre>
<pre id="l11"><span class="adr"></span>	                </pre>
<pre id="l12"><span class="adr">8004</span>	AF              	<span id="code12">xra	a		</span><span class="cmt">; –æ—á–∏—Å—Ç–∏—Ç—å –∞–∫–∫—É–º—É–ª—è—Ç–æ—Ä</span></pre>
<pre id="l13"><span class="adr">8005</span>	21 00 40        	<span id="code13">lxi	h, 4000h	</span><span class="cmt">; –ø–æ–º–µ—Å—Ç–∏—Ç—å –≤ HL –∞–¥—Ä–µ—Å –Ω–∞—á–∞–ª–∞ —ç–∫—Ä–∞–Ω–Ω–æ–π –æ–±–ª–∞—Å—Ç–∏</span></pre>
<pre id="l14"><span class="adr">8008</span>	16 00           	<span id="code14">mvi	d, 0		</span><span class="cmt">; –æ—á–∏—Å—Ç–∏—Ç—å —Ä–µ–≥–∏—Å—Ç—Ä D –º–æ–∂–Ω–æ –±—ã–ª–æ –±—ã –Ω–∞ –æ–¥–∏–Ω –±–∞–π—Ç –∫–æ—Ä–æ—á–µ</span></pre>
<pre id="l15"><span class="adr"></span>	                </pre>
<pre id="l16"><span class="adr">800A</span>	01 00 20        	<span id="code16">lxi	b, 256*32	</span><span class="cmt">; —ç–∫—Ä–∞–Ω –≤ "–û–∫–µ–∞–Ω–µ" —Å–æ—Å—Ç–æ–∏—Ç –∏–∑ –≤–µ—Ä—Ç–∏–∫–∞–ª—å–Ω—ã—Ö —Å—Ç–æ–ª–±–∏–∫–æ–≤ –ø–æ 256 –±–∞–π—Ç –≤ –∫–∞–∂–¥–æ–º</span></pre>
<pre id="l17"><span class="adr">800D</span>	72              <span class="l" id="label17">Loop</span><span id="code17">:	mov	m, d		</span><span class="cmt">; –∑–∞–ø–æ–ª–Ω—è–µ–º —ç–∫—Ä–∞–Ω —Ç—Ä—É—Ö–æ–π</span></pre>
<pre id="l18"><span class="adr">800E</span>	14              	<span id="code18">inr	d		</span><span class="cmt">; –ø—Ä–∏–º–µ—Ä–Ω–æ –Ω–∞ —Ç—Ä–µ—Ç—å</span></pre>
<pre id="l19"><span class="adr">800F</span>	23              	<span id="code19">inx 	h		</span><span class="cmt">; –¥–≤–∏–≥–∞—Ç—å—Å—è –¥–∞–ª—å—à–µ</span></pre>
<pre id="l20"><span class="adr"></span>	                </pre>
<pre id="l21"><span class="adr">8010</span>	0B              	<span id="code21">dcx 	b		</span><span class="cmt">; –≤–Ω–∏–º–∞—Ç–µ–ª—å–Ω–æ —Å–ª–µ–¥–∏–º –∑–∞ –¥–ª–∏–Ω–æ–π —Å—Ç–æ–ª–±–∏–∫–∞</span></pre>
<pre id="l22"><span class="adr">8011</span>	78              	<span id="code22">mov	a, b		</span><span class="cmt">; –µ—Å–ª–∏ —Å—Ç–æ–ª–±–∏–∫ </span></pre>
<pre id="l23"><span class="adr">8012</span>	B1              	<span id="code23">ora	c		</span><span class="cmt">; –Ω–µ –¥–æ—Ä–∏—Å–æ–≤–∞–Ω</span></pre>
<pre id="l24"><span class="adr">8013</span>	C2 0D 80        	<span id="code24">jnz	Loop		</span><span class="cmt">; –ø—Ä–æ–¥–æ–ª–∂–∞–µ–º –ø–∏—Å–∞—Ç—å –≤ —ç–∫—Ä–∞–Ω —Ç—Ä—É—Ö—É</span></pre>
<pre id="l25"><span class="adr"></span>	                </pre>
<pre id="l26"><span class="adr">8016</span>	21 00 41        	<span id="code26">lxi	h, 4100h	</span><span class="cmt">; —Ü–≤–µ—Ç –∫–∞–∂–¥–æ–π —Ç–æ—á–∫–∏ –æ–ø—Ä–µ–¥–µ–ª—è–µ—Ç—Å—è –±–∏—Ç–∞–º–∏ —Å–æ—Å–µ–¥–Ω–∏—Ö —á–µ—Ä–µ–∑ 100h –±–∞–π—Ç</span></pre>
<pre id="l27"><span class="adr">8019</span>	3E 00           	<span id="code27">mvi	a, 00h		</span><span class="cmt">; –ø–µ—Ä–≤—ã–π —Å—Ç–æ–ª–±–∏–∫</span></pre>
<pre id="l28"><span class="adr">801B</span>	CD 53 80        	<span id="code28">call	Strip		</span><span class="cmt">; –æ–∫—Ä–∞—à–∏–≤–∞–µ–º –Ω—É–ª—ë–º</span></pre>
<pre id="l29"><span class="adr"></span>	                </pre>
<pre id="l30"><span class="adr">801E</span>	21 00 43        	<span id="code30">lxi	h, 4300h	</span><span class="cmt">; –≤—Ç–æ—Ä–æ–π —Å—Ç–æ–ª–±–∏–∫</span></pre>
<pre id="l31"><span class="adr">8021</span>	3E 0F           	<span id="code31">mvi	a, 0fh		</span><span class="cmt">; –æ–∫—Ä–∞—à–∏–≤–∞–µ–º –≤ –¥–≤–µ –ø–æ–ª–æ—Å—ã</span></pre>
<pre id="l32"><span class="adr">8023</span>	CD 53 80        	<span id="code32">call	Strip		</span><span class="cmt">; –¥–ª—è –∫—Ä–∞—Å–æ—Ç—ã</span></pre>
<pre id="l33"><span class="adr"></span>	                </pre>
<pre id="l34"><span class="adr">8026</span>	21 00 45        	<span id="code34">lxi	h, 4500h	</span><span class="cmt">; —Ç—Ä–µ—Ç—å–µ –æ–∫–Ω–æ</span></pre>
<pre id="l35"><span class="adr">8029</span>	3E 33           	<span id="code35">mvi	a, 33h		</span><span class="cmt">; –≤—ã—Ö–æ–¥–∏—Ç</span></pre>
<pre id="l36"><span class="adr">802B</span>	CD 53 80        	<span id="code36">call	Strip		</span><span class="cmt">; –∫ –æ–∫–µ–∞–Ω—É</span></pre>
<pre id="l37"><span class="adr"></span>	                </pre>
<pre id="l38"><span class="adr">802E</span>	21 00 47        	<span id="code38">lxi	h, 4700h	</span><span class="cmt">; —Ä–æ–≤–Ω—ã–º –≤–µ—Ç—Ä–æ–º</span></pre>
<pre id="l39"><span class="adr">8031</span>	3E 55           	<span id="code39">mvi	a, 55h		</span><span class="cmt">; –¥—ã—à–∏—Ç</span></pre>
<pre id="l40"><span class="adr">8033</span>	CD 53 80        	<span id="code40">call	Strip		</span><span class="cmt">; –æ–∫–µ–∞–Ω</span></pre>
<pre id="l41"><span class="adr"></span>	                </pre>
<pre id="l42"><span class="adr">8036</span>	21 00 49        	<span id="code42">lxi	h, 4900h	</span><span class="cmt">; –∞ –∑–∞ –Ω–∏–º</span></pre>
<pre id="l43"><span class="adr">8039</span>	3E FF           	<span id="code43">mvi	a,0ffh		</span><span class="cmt">; –¥–∏–∫–æ–≤–∏–Ω–Ω—ã–µ</span></pre>
<pre id="l44"><span class="adr">803B</span>	CD 53 80        	<span id="code44">call	Strip		</span><span class="cmt">; —Å—Ç—Ä–∞–Ω—ã</span></pre>
<pre id="l45"><span class="adr"></span>	                </pre>
<pre id="l46"><span class="adr"></span>	                				<span class="cmt">; –∏ –Ω–∏–∫—Ç–æ</span></pre>
<pre id="l47"><span class="adr"></span>	                				<span class="cmt">; –Ω–µ –≤–∏–¥–µ–ª</span></pre>
<pre id="l48"><span class="adr"></span>	                				<span class="cmt">; —ç—Ç–∏—Ö —Å—Ç—Ä–∞–Ω</span></pre>
<pre id="l49"><span class="adr"></span>	                </pre>
<pre id="l50"><span class="adr">803E</span>	AF              	<span id="code50">xra	a		</span><span class="cmt">; —ç–∫—Ä–∞–Ω —Ä–∞–∑–º–∞–ª–µ–≤–∞–Ω</span></pre>
<pre id="l51"><span class="adr">803F</span>	D3 C1           	<span id="code51">out	BANKING		</span><span class="cmt">; –≤–æ–∑–≤—Ä–∞—â–∞–µ–º –µ–≥–æ –Ω–∞ –º–µ—Å—Ç–æ</span></pre>
<pre id="l52"><span class="adr"></span>	                </pre>
<pre id="l53"><span class="adr">8041</span>	3C              <span class="l" id="label53">Key</span><span id="code53">:	inr	a		</span><span class="cmt">; –Ω–µ –ø—Ä–æ–ø–∞–¥–∞—Ç—å –∂–µ –∞–∫–∫—É–º—É–ª—è—Ç–æ—Ä—É</span></pre>
<pre id="l54"><span class="adr">8042</span>	E6 3F           	<span id="code54">ani	3fh		</span><span class="cmt">; –≤–æ–∑—å–º–µ–º –∏–∑ –Ω–µ–≥–æ —Ç—Ä–∏ –º–ª–∞–¥—à–∏—Ö –±–∏—Ç–∞ –ø–µ—Ä–µ–¥–Ω–µ–≥–æ –ø–ª–∞–Ω–∞</span></pre>
<pre id="l55"><span class="adr">8044</span>	F6 40           	<span id="code55">ori	40h		</span><span class="cmt">; –∏ —Ç—Ä–∏ —Å—Ç–∞—Ä—à–∏—Ö –±–∏—Ç–∞ —Ñ–æ–Ω–∞</span></pre>
<pre id="l56"><span class="adr">8046</span>	D3 E1           	<span id="code56">out	VIDEO		</span><span class="cmt">; –∏ –Ω–∞–º–∞–∂–µ–º –Ω–∞ –∫–∞–¥—Ä</span></pre>
<pre id="l57"><span class="adr"></span>	                </pre>
<pre id="l58"><span class="adr">8048</span>	CD 09 E0        	<span id="code58">call	0e009h		</span><span class="cmt">; –ø–æ–¥–æ–∂–¥–µ–º –≤–≤–æ–¥–∞ —Å–∏–º–≤–æ–ª–∞ —Å –∫–ª–∞–≤–∏–∞—Ç—É—Ä—ã</span></pre>
<pre id="l59"><span class="adr">804B</span>	FE 1B           	<span id="code59">cpi	1bh		</span><span class="cmt">; –µ—Å–ª–∏ —ç—Ç–æ –Ω–µ Esc,</span></pre>
<pre id="l60"><span class="adr">804D</span>	C2 41 80        	<span id="code60">jnz	Key		</span><span class="cmt">; –ø—Ä–æ–¥–æ–ª–∂–∏–º –∏–Ω—Ç–µ—Ä–∞–∫—Ç–∏–≤–Ω—É—é —Ä–∞—Å–∫—Ä–∞—Å–∫—É</span></pre>
<pre id="l61"><span class="adr"></span>	                </pre>
<pre id="l62"><span class="adr">8050</span>	C3 03 E0        	<span id="code62">jmp	0e003h		</span><span class="cmt">; —Ç–µ–ø–ª—ã–π —Å—Ç–∞—Ä—Ç "–ú–æ–Ω–∏—Ç–æ—Ä–∞"</span></pre>
<pre id="l63"><span class="adr"></span>	                </pre>
<pre id="l64"><span class="adr">8053</span>	77              <span class="l" id="label64">Strip</span><span id="code64">:	mov	m, a		</span><span class="cmt">; —ç—Ç–æ –ø–æ–¥–ø—Ä–æ–≥—Ä–∞–º–º–∞</span></pre>
<pre id="l65"><span class="adr">8054</span>	23              	<span id="code65">inx	h		</span><span class="cmt">; –Ω–µ–±—ã—Å—Ç—Ä–æ–≥–æ, –Ω–æ –≤–µ—Ä–Ω–æ–≥–æ</span></pre>
<pre id="l66"><span class="adr">8055</span>	15              	<span id="code66">dcr	d		</span><span class="cmt">; —Ä–∏—Å–æ–≤–∞–Ω–∏—è —Å—Ç–æ–ª–±–∏–∫–∞</span></pre>
<pre id="l67"><span class="adr">8056</span>	C2 53 80        	<span id="code67">jnz	Strip		</span><span class="cmt">; —Å –∑–∞–¥–∞–Ω–Ω–æ—é –¥–ª–∏–Ω–æ—é</span></pre>
<pre id="l68"><span class="adr"></span>	                </pre>
<pre id="l69"><span class="adr">8059</span>	C9              	<span id="code69">ret			</span><span class="cmt">; –ö –û –ù –ï –¶</span></pre>
<pre id="l70"><span class="adr"></span>	                </pre>
<pre>Labels:</pre><div class="hordiv"></div><pre class="labeltable"><span class='t1' onclick="return gotoLabel('banking');">banking                 00C1</span><span class='t1' onclick="return gotoLabel('key');">key                     8041</span><span class='t1' onclick="return gotoLabel('loop');">loop                    800D</span><span class='t2' onclick="return gotoLabel('strip');">strip                   8053</span><br/><span class='t1' onclick="return gotoLabel('video');">video                   00E1</span></pre><div>&nbsp;</div><pre>Memory dump:</pre><div class="hordiv"></div><pre class="d0"> </pre><br/><pre class="d1">8000:  3E 01 D3 C1 AF 21 00 40-16 00 01 00 20 72 14 23  >....!.@.....r.#</pre><br/><pre class="d0">8010:  0B 78 B1 C2 0D 80 21 00-41 3E 00 CD 53 80 21 00  .x....!.A>..S.!.</pre><br/><pre class="d1">8020:  43 3E 0F CD 53 80 21 00-45 3E 33 CD 53 80 21 00  C>..S.!.E>3.S.!.</pre><br/><pre class="d0">8030:  47 3E 55 CD 53 80 21 00-49 3E FF CD 53 80 AF D3  G>U.S.!.I>..S...</pre><br/><pre class="d1">8040:  C1 3C E6 3F F6 40 D3 E1-CD 09 E0 FE 1B C2 41 80  .<.?.@........A.</pre><br/><pre class="d0">8050:  C3 03 E0 77 23 15 C2 53-80 C9                    ...w#..S........</pre><br/></body></html>